tests/test_json_to_parquet.py

```diff
@@ -0,0 +1,531 @@
+                table_key=glue_nested_table_name,
+                table_name=glue_nested_table_name,
+                processed_tables=tables_with_index,
+                unprocessed_tables=nested_table_relationalized,
+                glue_context=glue_context
+        )
+        assert (
+                set(tables_with_index[glue_nested_table_name].schema.fieldNames()) ==
+                set(["GlobalKey", "ArrayOfObjectsField", "ObjectField_filename",
+                     "ObjectField_timestamp", "export_end_date"])
+        )
+        table_key = f"{glue_nested_table_name}_ArrayOfObjectsField"
+        tables_with_index[table_key] =  json_to_parquet.add_index_to_table(
+                table_key=table_key,
+                table_name=glue_nested_table_name,
+                processed_tables=tables_with_index,
+                unprocessed_tables=nested_table_relationalized,
+                glue_context=glue_context
+        )
+        assert (
+                set(tables_with_index[table_key].schema.fieldNames()) ==
+                set(['id', 'index', 'filename', 'timestamp', 'GlobalKey'])
+        )
+        child_table_with_index = (tables_with_index[table_key]
+              .toPandas()
+              .sort_values("GlobalKey")
+              .reset_index(drop=True)
+              .drop(columns=["id", "index"]))
+        print("Child table with index:")
+        print(child_table_with_index)
+        correct_df = pandas.DataFrame({
+            "filename": ["test.json"],
+            "timestamp": ["999"],
+            "GlobalKey": ["123456789"]
+        })
+        print("Correct table:")
+        print(correct_df)
+        for col in child_table_with_index.columns:
+            print(col)
+            print(child_table_with_index[col].values == correct_df[col].values)
+            assert all(child_table_with_index[col].values == correct_df[col].values)
+
+    def test_write_table_to_s3(self, artifact_bucket, namespace, glue_database_name,
+                               glue_flat_table_name, glue_context, parquet_key_prefix):
+        flat_table = json_to_parquet.get_table(
+                table_name=glue_flat_table_name,
+                database_name=glue_database_name,
+                glue_context=glue_context
+        )
+        parquet_key = os.path.join(
+                    parquet_key_prefix,
+                    "dataset_TestFlatDataType"
+        )
+        json_to_parquet.write_table_to_s3(
+                dynamic_frame=flat_table,
+                bucket=artifact_bucket,
+                key=parquet_key,
+                glue_context=glue_context,
+                workflow_run_id="testing"
+        )
+        s3_client = boto3.client("s3")
+        parquet_dataset = s3_client.list_objects_v2(
+                Bucket=artifact_bucket,
+                Prefix=parquet_key)
+        assert parquet_dataset["KeyCount"] > 0
+
+    def test_archive_existing_datasets(self, glue_context, artifact_bucket,
```

Comment:
For test readability, it'd be ideal if these are split into separate test cases:

* test_archive_existing_datasets_non_existent
* test_archive_existing_datasets_existent
* test_archive_existing_datasets_delete_upon_completion
* test_archive_existing_datasets_diff_workflow_id (appends)

This is true for many of these. It's not a hard and fast rule, but each test should ideally test a particular condition of a feature.  By doing that, it's also easier to know which part of the code base failed when a particular test fails.

Reply:
I split these up.